packages/next-swc/crates/next-core/js/src/entry/app-renderer.tsx

```diff
@@ -245,7 +245,7 @@ async function runOperation(renderData: RenderData) {
     cssModules: {},
   }
   const req: IncomingMessage = {
-    url: renderData.url,
+    url: renderData.originalUrl,
```

Comment:
This might break the `_next/404` logic I set up to force rewrites to redirect to 404 pages.

Reply:
Do we have a test case for that?`